cinemy/cinema/viewset.py

```diff
@@ -12,10 +12,12 @@
 from rest_framework.permissions import IsAuthenticated
 
 
-class MovieCurrentlyPlayingViewSet(viewsets.ReadOnlyModelViewSet):
+class ReadOnlyModelViewSetWithAuth(viewsets.ReadOnlyModelViewSet):
```

Comment:
If you want to avoid duplicated code, use Mixins. `ReadOnlyModelViewSetWithAuth` doesn't really define a viewset; it is more a plugin / helping class for the following ones. Declare a mixin object with no inheritance and inherit both let's say `CustomerAuthMixin` (choose a name you like) and `ReadOnlyModelViewSet` on the child class (helps also with visibility to keep them separated).

Sometimes it might help to keep attributes explicitly declared on the class (for visibility), depending on the project, on how many endpoints you have declared etc. But it's fine either way.

Reply:
Done